### ISSUE TITLE
Dragon's Crown RPCN multiplayers failed to connect

### ISSUE THREAD

**Issue by JumpingY** [RPCS3/rpcs3#15283]
### Quick summary

![QQ截图20240307230447](https://github.com/RPCS3/rpcs3/assets/54348327/dc6db1fa-3913-415b-b8e8-c05466b78789)
![QQ截图20240307230453](https://github.com/RPCS3/rpcs3/assets/54348327/9e2431aa-3a87-4b07-a499-457574996eab)
![QQ截图20240307230507](https://github.com/RPCS3/rpcs3/assets/54348327/683c90d1-c4ab-4a2c-9257-4634a5a57833)
![QQ截图20240307230430](https://github.com/RPCS3/rpcs3/assets/54348327/879a2388-46ff-4710-b848-249bc275b930)

### Details

Local RPCN service with LAN connection.
Add friends to each other and see each other in the room list
RPCN v1.2.1

### Attach a log file

[RPCS3.log](https://github.com/RPCS3/rpcs3/files/14526438/RPCS3.log)

### System configuration

OS:Windows 10
GPU:GTX1660
CPU:AMD R7 3700X

**Comment by Ninetime** [comment]
Did you try connecting after starting a map? Your screenshot looks like it's just outside the town

**Comment by JumpingY** [comment]
> Did you try connecting after starting a map? Your screenshot looks like it's just outside the town

Yes, I have tried, but unfortunately, the problem still persists.In the screenshot, we tried to connect to the arena online.It is worth mentioning that occasionally one can successfully enter and complete a game, but reconnecting after exiting the room will continue to fail

**Comment by SoHaiNp** [comment]
The same issue happened to me... I tried to join a friend's room and we both were online on RPCN...

I was able to see my friend's room available to join at the "join friend's room" at the online lobby... But when i click on his room nothing happens, just a "connecting." message on the screen...

I recall that the dots after the "connecting" word should be looping when trying to connect when i used to play on PS3... I also tried to use the "join random room" option and tried to join his room while he was in the midle of a map, but nothing really worked.

If somebody needs any more information, tests or logs let me know, i would be glad to be able to play online this game again through my computer ^^

**Comment by grimm00** [comment]
I'm having the same issue as @SoHaiNp , where there is just a message that says "Connecting." when trying to establish a connection (even if my partner is already in a dungeon). I've never played this game before, but I also assumed that there was supposed to be a subtle animation of the dots after "Connecting" appearing in a way to show that connectivity is being attempted, which is not here. It'll hang on this message forever without ever showing any error. I've attached my own log.

The matchmaking seems to stop abruptly around line 22349 (~Timestamp 09:09:166)

[RPCS3.log](https://github.com/user-attachments/files/21026677/RPCS3.log)

**Comment by grimm00** [comment]
Okay so I may have figured out the problem. 

I think this might have something to do with a UDP Port conflict. The software seems to be hard-coded to use UDP Port 3658 for the host (in the stables) and the client (whoever's joining into the game). 

Unfortunately for your network, it's rightfully confused, like um why am I trying to open a connection up for myself?

So I figured out that I had to explicitly add a setting to the game's config file under the "Net" section: 
- For myself, I added "**UDP Port: 3659**" (pictured below). 
- On the other device, I added "**UDP Port: 3658**" to its config file.
- I also unchecked **UPNP Enabled** for both devices (or set to _**false**_ in config file). 

This got it working both ways (there is still slight connection issues, but usually if someone abruptly disconnect and/or we have to reestablish connection in the same game section). 

**Comment:**  While this has worked for me, I'm not sure this is the intended way it's supposed to work. I"m sure there is supposed to be some dynamic way to make this happen, so be cautious and **back up your config** just in case. 

**Note:** I'm using Linux (Steam Deck), so I'm not sure exactly where your file folders are, but it should be named "**config_BLUS30767.yml**"

![Image](https://github.com/user-attachments/assets/ca8ce4e4-f893-4056-80da-462fc3643aba)

**Comment by JumpingY** [comment]
[RPCS3.log.gz](https://github.com/user-attachments/files/25162482/RPCS3.log.gz)
update logs

**Comment by Ninetime** [comment]
@RipleyTom Do you see anything in the logs that might cause the disconnect described?
